src/DDTrace/Integrations/Mysqli.php

```diff
@@ -0,0 +1,422 @@
+            /** @var \DDTrace\Span $span */
+            $span = $scope->getSpan();
+            Mysqli::setConnectionInfo($span, $this);
+
+            try {
+                $statement = $this->prepare($query);
+                $host_info = Mysqli::extractHostInfo($this);
+                ObjectKVStore::put($statement, 'host_info', $host_info);
+                Mysqli::storeQuery($statement, $query);
+                return $statement;
+            } catch (\Exception $e) {
+                $span->setError($e);
+                throw $e;
+            } finally {
+                $scope->close();
+            }
+        });
+
+        // bool mysqli::commit ([ int $flags [, string $name ]] )
+        dd_trace('mysqli', 'commit', function () {
+            $args = func_get_args();
+            $resource = Mysqli::retrieveQuery($this, 'mysqli.commit');
+            $scope = Mysqli::initScope('mysqli.commit', $resource);
+            /** @var \DDTrace\Span $span */
+            $span = $scope->getSpan();
+            Mysqli::setConnectionInfo($span, $this);
+
+            if (isset($args[1])) {
+                $span->setTag('db.transaction_name', $args[1]);
+            }
+
+            try {
+                return $this->commit(...$args);
+            } catch (\Exception $e) {
+                $span->setError($e);
+                throw $e;
+            } finally {
+                $scope->close();
+            }
+        });
+
+        // bool mysqli_stmt::execute ( void )
+        dd_trace('mysqli_stmt', 'execute', function () {
+            $resource = Mysqli::retrieveQuery($this, 'mysqli_stmt.execute');
+            $scope = Mysqli::initScope('mysqli_stmt.execute', $resource);
+            /** @var \DDTrace\Span $span */
+            $span = $scope->getSpan();
+
+            try {
+                return $this->execute();
+            } catch (\Exception $e) {
+                $span->setError($e);
+                throw $e;
+            } finally {
+                $scope->close();
+            }
+        });
+
+        // bool mysqli_stmt::execute ( void )
+        dd_trace('mysqli_stmt', 'get_result', function () {
+            $resource = Mysqli::retrieveQuery($this, 'mysqli_stmt.get_result');
+            $scope = Mysqli::initScope('mysqli_stmt.get_result', $resource);
+            /** @var \DDTrace\Span $span */
+            $span = $scope->getSpan();
+
+            try {
+                $result = $this->get_result();
+                ObjectKVStore::propagate($this, $result, 'host_info');
+                ObjectKVStore::put($result, 'query', $resource);
+                return $result;
+            } catch (\Exception $e) {
+                $span->setError($e);
+                throw $e;
+            } finally {
+                $scope->close();
+            }
+        });
+
+        // Procedural fetch methods
+        Mysqli::traceProceduralFetchMethod('mysqli_fetch_all');
```

Comment:
We _can_ however use `self` on all these lines as well as the `Mysqli::traceConstructorFetchMethod()` calls bellow. :)

```suggestion
        self::traceProceduralFetchMethod('mysqli_fetch_all');
```

Reply:
This is very correct 👍 doing it right now!